doc/feature/spikeReport.md

```diff
@@ -0,0 +1,333 @@
+         *        UNDEFINED_TIMESTAMP, it will be considered as infinite.
+         */
+        boost::future< Spikes > read( float min );
+
+        /**
+         * Read all available spikes.
+         *
+         * This is a convenience function that behaves like read( -inf ) if
+         * getCurrentTime() is undefined and read( getCurrentTime( )) otherwise.
+         */
+        boost::future< Spikes > read();
+
+        /**
+         * Read spikes until getCurrentTime() == max, the end of the report is
+         * reached or the report closed.
+         *
+         * This function is very similar to the normal read, but instead of
+         * specifying a lower bound of getCurrentTime at return, it specifies
+         * a requested strict value. The preconditions and postconditions
+         * are the same except for those involving the parameter min which
+         * become:
+         *
+         * Precondition: max > getCurrentTime()
+         * Postcondition: If r.getState() == OK, then r.getCurrentTime() == max
+         *
+         * @sa forward()
+         */
+        boost::future< Spikes > readUntil( float max );
+
+        /**
+         * Forward to a given absolute timestamp.
+         *
+         * In read mode this means skipping data forward until the timestamp
+         * is made current. In write mode for streams, consumers are notified
+         * about the progress.
+         *
+         * Preconditions:
+         * - toTimestamp >= getCurrentTime()
+         *
+         * Postconditions:
+         * Let:
+         *  - r be the SpikeReport
+         *  - f be the returned future by r.forward(toTimestamp)
+         * Then:
+         *  - After f.wait() returns r.getCurrentTime() == toTimestamp.
+         *  - The postconditions of read operations imply that this function
+         *    throws away the data previous to toTimestamp (or avoids reading
+         *    it at all if possible).
+         */
+        boost::future< void > forward( float toTimestamp );
+
+        enum State { OK = 0, ENDED = 1, FAILED = 2 };
+        /** @return The state after the last completed operation. */
+        State getState() const;
+
+       /**
+         * Write the given spikes to the output.
+         *
+         * Upon return getCurrenTime() is the greatest of all the spike times
+         * plus an epsilon..
+         *
+         * @param spikes A sorted collection of spikes. For every spike, its
+         *        timestamp t must be >= getCurrentTime().
+         */
+        void write( const Spikes& spikes );
+
+        /*
+         * Closes the report.
+         *
+         * In read mode this call forces the state to be changed to ENDED.
+         * Any call waiting in any future will be unblocked as if there was
+         * no more data to be read.
+         *
+         * In write mode with stream reports this signals the end of the
+         * report to the consumers.
+         */
+        void close();
```

Comment:
Explicit close is needed or dtor would be enough? Can a closed report be used again for even a different operation?